packages/core/src/directory.ts

```diff
@@ -63,74 +59,59 @@ export class DirectorySync implements DirectorySubscriber {
 
             await this.server.updateMetaData(org);
             await this.server.storage.save(org);
-
-            return newOrgMember;
         }
     }
 
-    async userUpdated(user: DirectoryUser, orgId: string, userId: string) {
+    async userUpdated(user: DirectoryUser, orgId: string, previousEmail?: string) {
         const org = (orgId && (await this.server.storage.get(Org, orgId))) || null;
         if (org && org.directory.syncMembers) {
-            let existingUser: OrgMember | null = null;
-            for (const member of org.members) {
-                if (existingUser) {
-                    continue;
-                }
-
-                const provisioningId = await getIdFromEmail(member.email);
+            // The existence of the `previousEmail` argument indicates that the member's
+            // email has changed, so we'll have to look up the member using the previous email
+            const existingMember = org.getMember({ email: previousEmail || user.email });
 
-                if (member.accountId === userId || member.id === userId || provisioningId === userId) {
-                    existingUser = member;
-                }
+            if (!existingMember) {
+                return;
             }
 
-            if (!existingUser) {
-                return;
+            // Changing the email address of a user is a little more
+            // involved than that. Let's disable it for now.
+
+            if (previousEmail && previousEmail !== user.email) {
+                throw new Err(ErrorCode.NOT_SUPPORTED, "Updating user emails is not supported at this time");
```

Comment:
Actually changing the email of a user is actually fairly complicated and currently not supported. Let's tackle this at a later time.